packages/daemon/src/host.js

```diff
@@ -107,15 +107,17 @@ export const makeHostMaker = ({
      * @param {string} [petName]
      */
     const store = async (readerRef, petName) => {
+      /** @type {import('./types.js').DeferredTasks<import('./types.js').ReadableBlobDeferredTaskParams>} */
+      const tasks = makeDeferredTasks();
+
       if (petName !== undefined) {
         assertPetName(petName);
+        tasks.push(identifiers =>
+          petStore.write(petName, identifiers.readableBlobFormulaIdentifier),
+        );
       }
 
-      const formulaIdentifier = await storeReaderRef(readerRef);
-
-      if (petName !== undefined) {
-        await petStore.write(petName, formulaIdentifier);
-      }
+      return (await incarnateReadableBlob(readerRef, tasks)).value;
```

Comment:
Easier to read with a named intermediate variable. Not necessary, but in general, we like to avoid deeply embedding `await` in expressions, in order to make its presence more obvious.